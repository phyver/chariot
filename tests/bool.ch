data bool where True | False : bool

val f True = False
  | f False = f True

val g True = False
  | g x = g True

:set expand_clauses true
val h True = False
  | h x = h True


