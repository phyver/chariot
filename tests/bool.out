>> option verbose is set to 1
>> inductive type bool was succesfully defined
>> Typing for f successful
>> the definition for f is complete
>> the definition f is provably correct

>> Typing for g successful
>> the definition for g is complete
!! warning: the definition g is NOT provably total (weight_bound: 2, depth_bound: 2)

>> option expand_clauses is set to true
>> Typing for h successful
>> the definition for h is complete
>> the definition h is provably correct

